### PR TITLE
auth/session API call and console logging out the response is having an error.

### PR DESCRIPTION
**Pull request by janie-lee-developer**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

I am calling an API call auth/session, and trying to console.log the response from the API call, but I am getting this error. It's saying I am trying to render an obj inside a jsx element. I don't think I am doing that. Also, this branch is up to date with the remote hot repo's beta branch. In case the error might be from some other issue.

## Related Tickets & Documents

#366 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by bdougie** [comment]
@janie-lee-developer canyou paste the error message here too?

File: src/components/PrimaryNav.tsx [hunk 1]
```diff
@@ -29,13 +29,19 @@ const StarTheRepo = (): JSX.Element => (
   </div>
 );
 
-const PrimaryNav = (): JSX.Element => {
+const PrimaryNav = async (): Promise<JSX.Element> => {
```
Comment (bdougie) [inline src/components/PrimaryNav.tsx:32]: This is the error. async is not needed is not ideal. Instead `useEffect`. Here is an example https://github.com/open-sauced/hot/blob/beta/src/components/RepoWrap.tsx#L48-L50

https://docs.react-async.com/

Comment (janie-lee-developer) [reply src/components/PrimaryNav.tsx]: Hi Brian! Thank you! I removed async on the component, PrimaryNav, and that error is gone now!

I am still having following 2 errors.
1. the API call is returning 401 status.
2.  npm run format fails, saying that I need to catch the error, so I added the try and error keywords, but It's still failing the npm run format... Could you please help?
Screen Shot 2022-10-12 at 3 52 10 PM
Screen Shot 2022-10-12 at 3 51 10 PM

Comment (bdougie) [reply src/components/PrimaryNav.tsx]: Following up from my DM. 

1. Check if you are passing the token or jwt in the call to the API - 401 unauthorized makes me assume not.
2. add a .catch to your function - similar to here: https://github.com/open-sauced/hot/blob/b801823659e89ef81f7eb14682901a20275b8990/src/hooks/useVotedRepos.ts#L18-L20

Comment (janie-lee-developer) [reply src/components/PrimaryNav.tsx]: Hi Brian, thank you so much for the help during the meeting! I am now getting 200 successfully, but for some reason, when I run npm run format, I am still getting an error saying that 'Promise must be awaited'.
Screen Shot 2022-10-13 at 2 08 22 PM
Screen Shot 2022-10-13 at 2 06 40 PM

Screen Shot 2022-10-13 at 2 06 50 PM

**Comment by janie-lee-developer** [comment]
Hi Brian, could you advise what I can do to fix the error that I am getting from npm run format? I just want to finish up this part before I move on to submissions API.

**Comment by janie-lee-developer** [comment]
@bdougie 
So I used the useEffect to useSession, but followings are changes I made.
1. @ useSession needed to use useSupabaseAuth, so I removed this and replaced with /lib/supabase.
2. added context folder at root level to create the global-state.tsx file.
3. created interfaces folder at root level to create the global-state-types.ts file.
4.  I removed the onboarded and waitlisted properties being added to the global context when useSession is ran.

Getting 404 error. 

Screen Shot 2022-10-18 at 4 06 10 PM

**Comment by 0-vortex** [comment]
> 1. @ useSession needed to use useSupabaseAuth, so I removed this and replaced with /lib/supabase.

This call doesn't need to be cached, it's safe to call every time in supabaseAuth

> 2. added context folder at root level to create the global-state.tsx file.
> 3. created interfaces folder at root level to create the global-state-types.ts file.
> 4. I removed the onboarded and waitlisted properties being added to the global context when useSession is ran.

This is not needed nor desired as part of this particular issue, since it is a big scope creep, creating a global state [1] and then caching that session data [2]. They are great things to do, when the users can accurately log in.

You can keep this draft open as it will ideally solve an additional 2 more issues, however the code itself needed for this is just 1 line fetching the `auth/session` api in supabaseAuth hook.

> Getting 404 error.
> 
> Screen Shot 2022-10-18 at 4 06 10 PM

That error says the avatar image for the user is missing, it's unrelated.

**Comment by bdougie** [comment]
> You can keep this draft open as it will ideally solve an additional 2 more issues, however, the code itself needed for this is just 1 line fetching the auth/session api in supabaseAuth hook.

@janie-lee-developer - open a new PR with the previous commit that just calls auth/session. As @0-vortex mentioned, leave this one open for solving the other things later, but for now. Just calling that endpoint is enough to move on to submissions.

**Comment by janie-lee-developer** [comment]
Okay! Understood! Thank you Ted and Brian. Will do!

File: src/components/PrimaryNav.tsx (hunk 1 above)
Comment (SunGoldTech) [reply src/components/PrimaryNav.tsx]: Hey Janie-lie, this error is as a result of execution of an async function without the await keyword... adding the `await` keyword to the `fetchAuthSection()` function call should fix. And on a second thought, the function should return the api's call response/error and not `console.log` response/error.

**Comment by bdougie** [comment]
Closing this now. I think we have done what is needed to unblock submissions and handle errors.

**Pull request closed without merging**